core/src/Dscp/Core/FairCV.hs

```diff
@@ -100,6 +102,16 @@ addProof educatorAddr blkHash proof fcv@(FairCV sAddr sName _) =
     mergeFairCVs fcv $
     singletonFCV sAddr sName educatorAddr blkHash proof
 
+-- | Make a FairCV from one private block.
+privateBlockToFairCV :: PrivateBlock -> Address -> (Address, Text) -> FairCVReady
+privateBlockToFairCV blk educator (student, studentName) =
+    let txs = blk ^. pbBody . pbbTxs
+        blkHash = hash (_pbHeader blk)
+        proof = mkMerkleProof (fromList txs) (fromList $ zipWith const [1..] txs)
```

Comment:
You would already build a Merkle tree from transactions list in `createPrivateBlock` at this point, and here you would do it again, while it's really unnecessary - because you create a proof for a _full_ tree, so it does not have any intermediate hashes, you only need to construct a bunch of empty `ProofBranch` nodes. 
This is not essential, but it would be nice to have such a function which creates a full `MerkleProof` from list. It will also get rid of `Maybe` and this ugly `error`. Making such a function will probably require generalizing `Dscp.Crypto.MerkleTree.nodeFromList` a bit.

Reply:
> You would already build a Merkle tree from transactions list in createPrivateBlock at this point, and here you would do it again, while it's really unnecessary - because you create a proof for a full tree, so it does not have any intermediate hashes, you only need to construct a bunch of empty ProofBranch nodes.

So instead I would better create a `MerkleProof` from transactions list directly?
Under empty `ProofBranch` nodes you mean ones with transactions, not just hashes?

> This is not essential, but it would be nice to have such a function which creates a full MerkleProof from list. It will also get rid of Maybe and this ugly error. Making such a function will probably require generalizing `Dscp.Crypto.MerkleTree.nodeFromList` a bit.

You mean, generalizing in a way which allows creating both `MerkleNode`s and `MerkleProof`s?

Reply:
>So instead I would better create a `MerkleProof` from transactions list directly?
Under empty `ProofBranch` nodes you mean ones with transactions, not just hashes?

`MerkleProof` has `ProofLeaf`s, which contain actual values of a list, and `ProofPruned`s, which contain only hashes. If a Merkle proof includes all the items in the original array, there are no `ProofPruned`s, only `ProofLeaf`s and `ProofBranch`es. And yes, it's easy to create such a proof from an original array, you won't even need to hash anything.

>You mean, generalizing in a way which allows creating both `MerkleNode`s and `MerkleProof`s?

Yes